tools/RAiDER/utilFcns.py

```diff
@@ -814,17 +796,21 @@ def write2NETCDF4core(nc_outfile, dimension_dict, dataset_dict, tran, mapping_na
         )
         # variable made, now add attributes
 
-        var.setncattr('grid_mapping_name', grid_mapping)
-        var.setncattr('straight_vertical_longitude_from_pole', srs.GetProjParm('central_meridian'))
-        var.setncattr('false_easting', srs.GetProjParm('false_easting'))
-        var.setncattr('false_northing', srs.GetProjParm('false_northing'))
-        var.setncattr('latitude_of_projection_origin', np.sign(srs.GetProjParm('latitude_of_origin')) * 90.0)
-        var.setncattr('latitude_of_origin', srs.GetProjParm('latitude_of_origin'))
-        var.setncattr('semi_major_axis', float(srs.GetAttrValue('GEOGCS|SPHEROID', 1)))
-        var.setncattr('scale_factor_at_projection_origin', 1)
-        var.setncattr('inverse_flattening', float(srs.GetAttrValue('GEOGCS|SPHEROID', 2)))
-        var.setncattr('spatial_ref', srs.ExportToWkt())
-        var.setncattr('spatial_proj4', srs.ExportToProj4())
+        for k, v in crs.to_cf().items():
+            var.setncattr(k, v)
+
+        # TODO - delete this once we know part above works as expected
+        # var.setncattr('grid_mapping_name', grid_mapping)
+        # var.setncattr('straight_vertical_longitude_from_pole', srs.GetProjParm('central_meridian'))
+        # var.setncattr('false_easting', srs.GetProjParm('false_easting'))
+        # var.setncattr('false_northing', srs.GetProjParm('false_northing'))
+        # var.setncattr('latitude_of_projection_origin', np.sign(srs.GetProjParm('latitude_of_origin')) * 90.0)
+        # var.setncattr('latitude_of_origin', srs.GetProjParm('latitude_of_origin'))
+        # var.setncattr('semi_major_axis', float(srs.GetAttrValue('GEOGCS|SPHEROID', 1)))
+        # var.setncattr('scale_factor_at_projection_origin', 1)
+        # var.setncattr('inverse_flattening', float(srs.GetAttrValue('GEOGCS|SPHEROID', 2)))
+        var.setncattr('spatial_ref', crs.to_wkt())
+        var.setncattr('spatial_proj4', crs.to_proj4())
         var.setncattr('spatial_epsg', epsg)
```

Comment:
This type of georeferenced writing to a netcdf can be greatly simplified with https://corteva.github.io/rioxarray/stable/

Reply:
I'm not sold on rioxarray. Will let team decide on it. `pyproj.to_cf` works well as it is the intended usage. I will leave changes to netcdf creation to folks who are owners of this capability.